core/src/test/scala/org/apache/spark/storage/BlockManagerSuite.scala

```diff
@@ -2093,6 +2093,9 @@ class BlockManagerSuite extends SparkFunSuite with Matchers with BeforeAndAfterE
       Seq("hostC", "hostB", "hostD").sorted)
     assert(master.getShufflePushMergerLocations(4, Set.empty).map(_.host).sorted ===
       Seq("hostB", "hostA", "hostC", "hostD").sorted)
+    master.removeShufflePushMergerLocation("hostA")
+    assert(master.getShufflePushMergerLocations(4, Set.empty).map(_.host).sorted ===
+      Seq("hostB", "hostC", "hostD").sorted)
```

Comment:
Can we extend this UT to verify the driver host is excluded? It will ensure that any future changes will not change this.

Reply:
I tried doing that, currently the BlockManagerMaster is mocked, therefore it doesn't register due to which it won't be added as part of `blockManagerIdByExecutor`. If not the test should have failed in the first place itself, as we are explicitly having tests checking for the hosts. But I will see if there is any other way add tests for this thing.

Reply:
Sounds fine.

Reply:
@Ngone51 Any suggestions? or is this fine as it is? I don't think there is any other way if not we need to significantly refactor the tests or something.

Reply:
It's fine as it is.